test/libsolidity/semanticTests/variables/transient_state_variable_initialization.sol

```diff
@@ -0,0 +1,16 @@
+contract C {
+    uint transient public x = 2;
+
+    constructor() {
+        x = 1;
+    }
+
+}
```

Comment:
```suggestion
contract C {
    uint transient public x = 2;

    constructor() {
        x = 1;
    }
    
    // check that this is identity
    function test(uint arg) public returns (uint) {
      x = arg;
      return this.x();
    }

}
```

Reply:
Has this been added? I don't see any test that would resemble this.

Also, some more coverage for transient variables in creation context (i.e. constructor) would not hurt.

Reply:
This is an old comment before we forbade initializing transient state variables in place, i.e. 
```
contract C {
  transient uint256 x = 2;
}
```

As far as constructor assignment is concerned, I could have sworn we had one in the legacy implementation PR, but that was a syntax test, so I'll add a semantic one as well.